src/test/java/study/StringTest.java

```diff
@@ -1,13 +1,60 @@
 package study;
 
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public class StringTest {
     @Test
     void replace() {
         String actual = "abc".replace("b", "d");
         assertThat(actual).isEqualTo("adc");
     }
+
+    @Test
+    void split() {
+        //given
+        String actual = "1,2";
+
+        //when1
+        String[] spltedStrList = actual.split(",");
+        //then1
+        assertThat(spltedStrList).contains("1");
+        assertThat(spltedStrList).contains("2");
+
+        //given
+        String actual2 = "1";
+
+        //when2
+        String[] spltedStrList2 = actual2.split(",");
+        //then2
+        assertThat(spltedStrList2).containsExactly("1");
+    }
```

Comment:
해당 테스트 케이스에서 두 가지 상황을 테스트 하려고 하고 있어요.

두 가지 상황을 하나의 테스트 케이스에서 다룬다면 어떤 문제가 있을까요?
- 만약 제가 아래와 같은 코드를 작성 했다고 한다면 두번째 케이스는 정상적으로 테스트가 될 수 있을까요?

테스트 케이스를 왜 분리 해야 하는지 Stack Overflow에서 설명 해준 게시글이 있어서 한 번 읽어보시고 참고 해보시면 좋을 것 같아요👏
[go to stackoverflow](https://stackoverflow.com/questions/235025/why-should-unit-tests-test-only-one-thing)
```suggestion
    @Test
    void split() {
        //given
        String actual = "1,2";

        //when1
        String[] spltedStrList = actual.split(",");
        //then1
        assertThat(spltedStrList).contains("1");
        assertThat(spltedStrList).contains("2");
        assertThat(spltedStrList).contains("3");

        //given
        String actual2 = "1";

        //when2
        String[] spltedStrList2 = actual2.split(",");
        //then2
        assertThat(spltedStrList2).containsExactly("1");
    }
```

Reply:
안녕하세요 주환님!

우선 코멘트 감사합니다.
저도 작성하면서 고민했던 부분인데 Stack Overflow 참고 도움되었습니다.

적어주신 코드에서 첫번째 상황에서 에러가 난다면 두번째 상황에서는 테스트를 진행할 수 없어 제대로된 테스트가 불가능하네요.

상황이 여러개라면 분리해서 독립적인 테스트를 만드는게 맞는 것 같습니다.  😄